content/blog/2019/02/2019-02-22-serverless-jenkins.adoc

```diff
@@ -0,0 +1,126 @@
+
+----
+travis encrypt CF_PW=supersecret --add
+travis encrypt ERP_PW=alsosupersecret --add
+----
+
+This command will add a line to your `.travis.yml` file with the encrypted secret value. Be sure to commit this change. Also take note of the name of your variable, which must match the environment parameter, and your Jenkins configuration. Also take note of https://docs.travis-ci.com/user/best-practices-security/[this TravisCI document on secrets].
+
+We’ll also need to add a `jenkins.yml` file to our project. Here we need to configure two shared libraries which are required for the SAP S/4HANA Cloud SDK Pipeline, and the credentials that are required for our pipeline. Be sure not to put your secrets in plain text in here, but use the variables you used before via the travis cli tool. TravisCI will decrypt the password on the fly for you.
+
+----
+jenkins:
+  numExecutors: 10
+unclassified:
+  globallibraries:
+    libraries:
+    - defaultVersion: "master"
+      name: "s4sdk-pipeline-library"
+      retriever:
+        modernSCM:
+          scm:
+            git:
+              remote: "https://github.com/SAP/cloud-s4-sdk-pipeline-lib.git"
+    - defaultVersion: "master"
+      name: "piper-library-os"
+      retriever:
+        modernSCM:
+          scm:
+            git:
+              remote: "https://github.com/SAP/jenkins-library.git"
+credentials:
+  system:
+    domainCredentials:
+      - credentials:
+          - usernamePassword:
+              scope: GLOBAL
+              id: "MY-ERP"
+              username: MY_USER
+              password: ${ERP_PW}
+          - usernamePassword:
+              scope: GLOBAL
+              id: "cf"
+              username: P12344223
+              password: ${CF_PW}
+----
+
+You might add more configuration to this file https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/README.md#initial-configuration[as you need it].
+
+Commit both files to your repo and push. If the travis build works, you’ll see the build integration on GitHub.
+
+image:/images/post-images/2019-02-22/in-progress.png
+
+On travis, you can follow the progress of your build live, and get the full text log of your Jenkins build. If all went well, you will be greeted with a green build after a few minutes.
+
+image:/images/post-images/2019-02-22/log.png
+
+Congratulations. You’re running a serverless Jenkins build with all the qualities checked by the SAP S/4HANA Cloud SDK Pipeline, without hosting your own Jenkins instance.
+
+Keep in mind this is a proof of concept at this point. The serverless Jenkins ecosystem is currently evolving, and neither Jenkinsfile runner, nor Configuration as Code are in a mature state as of February 2019. One downside of this approach is that we lose the Jenkins user interface, so we can’t see our pipeline in blue ocean, and we don’t get the nice build summary. We can get the whole log output from TravisCI, so this can be mitigated, but this is arguable not the best user experience.
```

Comment:
"as of February 2019" seems a bit redundant given it's a blog post published in February. 🤷‍♂️

Reply:
True, still I think the emphasis is good to have here as this is really not a statement that might be true a few months from now.